cgrok/grok_match.c

```diff
@@ -52,8 +52,7 @@ int grok_match_walk_next(const grok_match_t *gm,
   }
 
   *namelen = gct->name_len;
-  *name = malloc(*namelen);
-  memcpy(*name, gct->name, *namelen);
+  *name = gct->name;
```

Comment:
This was a memory leak on the hot path - every named group that matched would leak a string.